flow/storage/orientdb/orientdb.go

```diff
@@ -43,214 +42,188 @@ type Storage struct {
-			"Type": flow.ICMP.Type.String(),
-			"Code": flow.ICMP.Code,
-			"ID":   flow.ICMP.ID,
-		}
-	}
-	if flow.Transport != nil {
-		flowDoc["Transport"] = orient.Document{
-			"Protocol": flow.Transport.Protocol.String(),
-			"A":        flow.Transport.A,
-			"B":        flow.Transport.B,
-			"ID":       flow.Transport.ID,
-		}
+func metricToDoc(rid string, m *flow.FlowMetric) *metricDoc {
+	return &metricDoc{
+		Class:     "FlowMetric",
+		Type:      "d",
+		Flow:      rid,
+		ABBytes:   m.ABBytes,
+		ABPackets: m.ABPackets,
+		BABytes:   m.BABytes,
+		BAPackets: m.BAPackets,
+		Start:     m.Start,
+		Last:      m.Last,
 	}
-	return flowDoc
 }
 
-func documentToFlow(document orient.Document) (flow *flow.Flow, err error) {
-	if err = mapstructure.WeakDecode(document, &flow); err != nil {
-		return nil, err
+func (m *metricDoc) metric() *flow.FlowMetric {
+	return &flow.FlowMetric{
+		ABBytes:   m.ABBytes,
+		ABPackets: m.ABPackets,
+		BABytes:   m.BABytes,
+		BAPackets: m.BAPackets,
+		Start:     m.Start,
+		Last:      m.Last,
 	}
-	return
 }
 
-func documentToMetric(document orient.Document) (common.Metric, error) {
-	flowMetric := new(flow.FlowMetric)
-	if err := mapstructure.WeakDecode(document, flowMetric); err != nil {
-		return nil, err
+func rawpacketToDoc(rid string, linkType layers.LinkType, r *flow.RawPacket) *rawpacketDoc {
+	return &rawpacketDoc{
+		Class:     "FlowRawPacket",
+		Type:      "d",
+		Flow:      rid,
+		LinkType:  linkType,
+		Index:     r.Index,
+		Timestamp: r.Timestamp,
+		Data:      r.Data,
 	}
-
-	return flowMetric, nil
 }
 
-func documentToRawPacket(document orient.Document) (*flow.RawPacket, layers.LinkType, error) {
-	// decode base64 by hand as the json decoder used by orient db client just see a string
-	// and can not know that this is a array of byte as it only see interface{} as value
-	var err error
-	if document["Data"], err = base64.StdEncoding.DecodeString(document["Data"].(string)); err != nil {
-		return nil, layers.LinkType(0), err
-	}
-
-	rawpacket := new(flow.RawPacket)
-	if err = mapstructure.WeakDecode(document, rawpacket); err != nil {
-		return nil, layers.LinkType(0), err
+func (r *rawpacketDoc) rawpacket() *flow.RawPacket {
+	return &flow.RawPacket{
+		Index:     r.Index,
+		Timestamp: r.Timestamp,
+		Data:      r.Data,
 	}
-
-	l, err := document["LinkType"].(json.Number).Int64()
-	if err != nil {
-		return nil, layers.LinkType(0), err
-	}
-
-	return rawpacket, layers.LinkType(l), nil
 }
 
 // StoreFlows pushes a set of flows in the database
 func (c *Storage) StoreFlows(flows []*flow.Flow) error {
 	// TODO: use batch of operations
 	for _, flow := range flows {
-		flowDoc, err := c.client.Upsert(flowToDocument(flow), "UUID")
+		fd := flowToDoc(flow)
+		raw, err := json.Marshal(fd)
 		if err != nil {
-			logging.GetLogger().Errorf("Error while pushing flow %s: %s\n", flow.UUID, err.Error())
+			logging.GetLogger().Errorf("Error while pushing flow %s: %s", flow.UUID, err)
```

Comment:
nit: return fmt.Errorf("Error while pushing flow %s: %s", flow.UUID, err)
and log in the caller